src/library-aux/scala/Any.scala

```diff
@@ -138,15 +138,107 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to treat the receiver object to be of type `T0`.
+   *
+   *  This operation should be considered unsafe and the outcome **undefined**. On a language level it merely 
+   *  tells the compiler to forget any type information it has about the receiver object and treat it as if it
+   *  had type `T0`. How the compiler manages to transition from the type of the receiver object to `T0` is 
+   *  **undefined**.
+   *
+   *  Since the outcome is undefined, it is an implementation detail of the compiler backend whether this call 
+   *  results in an operation in the target bytecode (e.g. Java bytecode) or not. 
+   *  Some backends might emit a conversion, a runtime check or replace the receiver object with a default value.
```

Comment:
As I mention below, the *conversion* cases are actually specified, as special cases. The *casts* result in an unspecified amount of *checks*, but they never convert.

Reply:
note sure what you mean by they never convert. Targeting JDK they do convert as specified by the JLS. For instance: 1.1.asInstanceOf[Int] will narrow it to 1

Reply:
`1.1.asInstanceOf[Int]` is a *conversion* case, because the receiver's static type and the target type are both primitive types. It is *not* a cast.

Reply:
Sorry, I totally misread your first comment. I figured you want to say asInstanceOf only casts never converts. I agree on the definition of conversion and cast and would add asInstanceOf might as well insert boxing conversions not only conversions between primitive types